web-app/js/smartR/heatmapService.js

```diff
@@ -257,16 +261,45 @@ window.HeatmapService = (function(smartRHeatmap){
         console.log('service.runAnalysis', params);
         var defer = jQuery.Deferred();
 
-        function showD3HeatMap(data) {
-            downloadJsonFile(this.executionId, 'heatmap.json')
-                    .then(function(d) { smartRHeatmap.create(d); })
-                    .then(function() { defer.resolve(); });
+        function runAnalysisSuccess(data) {
+            var i = 2;
```

Comment:
We need two pieces of data -- the heatmap and the table. We only resolve the deferred after we get both. So it's sort of like using `$.when()`